### PR TITLE
Fix splitting by channel on the first step of flows

### PR DESCRIPTION
**Pull request by nicpottier**
Went a few different paths on this, but ultimately I think this is the most correct and simplest path. Basically if we don't have a msg.channel, we derive the outgoing channel for the contact when building the context.

### CODE REVIEW

**Comment by nicpottier** [comment]
@norkans7 plz review as you did the previous version..

File: temba/flows/models.py [hunk 1]
```diff
@@ -1206,15 +1206,23 @@ def value_wrapper(value):
...
 
+        channel_context = None
+
+        # Figure out what the outgoing channel would be for this contact if we have one and populate it
+        # (msg channel will take precedence below if there is one)
+        if contact:
+            _contact, contact_urn = Msg.resolve_recipient(self.org, self.created_by, contact, None)
+            scheme = contact_urn.scheme if contact_urn else TEL_SCHEME
+            channel = contact.org.get_send_channel(contact_urn=contact_urn, scheme=scheme)
+            channel_context = channel.build_message_context() if channel else None
```
Comment (ericnewcomer) [inline temba/flows/models.py]: Any reason not to do this in the ```elif contact:``` block below? Now sure how expensive this is, but seems we don't want to do it if we are just going to throw it away.

**Comment by ericnewcomer** [comment]
looks good!

**Comment by norkans7** [comment]
Yep

**Pull request merged**